composer/trainer/strategy.py

```diff
@@ -0,0 +1,245 @@
+            to find parameters to not expect gradients for. This is useful if there are some
+            parameters in the model that are not being trained.
+    """
+    if dist.is_available() and dist.is_initialized():
+        if any((p.requires_grad for p in module.parameters())):
+            log.debug('Wrapping model with DistributedDataParallel')
+            ddp_model = DistributedDataParallel(module, find_unused_parameters=find_unused_parameters)
+            return ddp_model
+        return module
+    if dist.is_available():
+        raise RuntimeError('Please call dist.initialize_dist() before calling ddp.prepare_module()')
+
+    raise RuntimeError('When the world size is > 1, ``torch.distributed`` must be used. However, it is '
+                       'not available in your installation of PyTorch. Please install or build PyTorch '
+                       'with distributed support.')
+
+
+def prepare_fsdp_module(model: torch.nn.Module, optimizers: torch.optim.Optimizer, fsdp_config: Dict[str, Any], precision: Precision, device: str) -> None:
+    sharding_map = {
+        'NO_SHARD': ShardingStrategy.NO_SHARD,
+        'SHARD_GRAD_OP': ShardingStrategy.SHARD_GRAD_OP,
+        'FULL_SHARD': ShardingStrategy.FULL_SHARD,
+    }
+    sharding_strategy = sharding_map[fsdp_config.get('sharding_strategy', 'FULL_SHARD').upper()]
+
+    cpu_offload = CPUOffload(offload_params=True) if fsdp_config.get('cpu_offload', False) else None
+    if cpu_offload is not None:
+        raise ValueError('FSDP CPU Offload not supported yet.')
+
+    def _get_torch_dtype(dtype: Union[Precision, str]):
+        dtype = dtype.value if isinstance(dtype, Precision) else dtype
+        if dtype in ['float32', 'torch.float32', 'fp32']:
+            return torch.float32
+        elif dtype in ['float16', 'torch.float16', 'half', 'fp16', 'amp', 'amp_fp16']:
+            return torch.float16
+        elif dtype in ['bfloat16', 'bfloat', 'torch.bfloat16', 'bf16', 'amp_bf16']:
+            return torch.bfloat16
+        else:
+            raise ValueError(f'Not sure how to convert dtype={dtype} to a torch dtype.')
+
+    if fsdp_config.get('mixed_precision', None):
+        param_dtype = _get_torch_dtype(fsdp_config['mixed_precision']['param_dtype'])
+        reduce_dtype = _get_torch_dtype(fsdp_config['mixed_precision']['reduce_dtype'])
+        buffer_dtype = _get_torch_dtype(fsdp_config['mixed_precision']['buffer_dtype'])
+    else:
+        param_dtype = torch.float32 if fsdp_config.get('fp32_master_weights', True) else _get_torch_dtype(precision)
+        reduce_dtype = _get_torch_dtype(precision)
+        buffer_dtype = param_dtype
+
+    mixed_precision = MixedPrecision(
+        param_dtype=param_dtype,
+        reduce_dtype=reduce_dtype,
+        buffer_dtype=buffer_dtype,
+    )
+    backward_prefetch = BackwardPrefetch.BACKWARD_POST
+
+    # We choose to not wrap the ComposerModel directly, but instead wrap any submodules
```

Comment:
Perhaps explain this decision a bit?

Could we use `model.get_named_children()` instead of `dir(model)` and filtering?